lib/fluent/plugin/in_sample.rb

```diff
@@ -118,14 +120,22 @@ def emit(num)
     end
 
     def generate
-      d = @sample[@sample_index]
+      if @renew_record
+        d = @sample[@sample_index].dup
+      else
+        d = @sample[@sample_index]
+      end
```

Comment:
The code will be simpler if this can be extracted into another method, such as `sample(index)`.